### DISCUSSION TITLE
Loading DWARF information in memory

### DISCUSSION THREAD

**Discussion by djipi** [davea42/libdwarf-code#194] [Q&A] [answered]
Hello,
I know it may sound quite old, but back in the days with the libdwarf-20210528 package, a function dwarf_elf_init allowed to load DWARF stuff in memory and an application could parse it from there.
Do there is a 'how-to' documentation to explain how to load the DWARF stuff in memory with the newest version?
I'm using Visual Studio 2017 and Windows 10.
Thank you.

**Comment by davea42** [comment, marked as answer]
The capability still exists.  The latest document (in html) is
www.prevanders.net/libdwarfdoc

The table of contents entry "Reading DWARF With No Object File Present"
is what you want to read.  In addition, those comments have a link to
a fully worked out example (in the See Also) with code that does exactly
what you are asking about.

There is little difference from the earlier versions.  Adding some fields
to the public  struct with its function-pointers (which were lacking
a few things) is the main change from 2021.

Hope this helps.

**Comment by davea42** [comment]
The fully worked out example is in the distribution as src/bin/dwarfexample/jitreader.c
and (like all the provided examples) the initial lines specify the code of jitreader.c
is public domain.  The html version does not show the first few lines, so see
the source itself of jitreader.c

**Comment by djipi** [comment]
Thank you for this information. I will give it a try.
